fms/distributed/tensorparallel.py

```diff
@@ -47,117 +44,30 @@ def apply_embedding_tp(par_mod: nn.Embedding, mod: nn.Embedding, world_size, ran
     # print(f"For rank {rank}, we have the following weights: Base weight {mod.weight} bias {mod.bias}; Par weight {par_mod.weight}, bias {par_mod.bias}")
 
 
-def _all_reduce(input_: torch.Tensor) -> torch.Tensor:
-    """All-reduce the input tensor across model parallel group."""
-    world_size = torch.distributed.get_world_size()
-
-    if world_size == 1:
-        return input_
-
-    return distfunc.all_reduce(input_, "sum", list(range(world_size)))
-
-
-# Fix #1 is porting the code changes in https://github.com/pytorch/pytorch/pull/108811
-@classmethod
-def wait_create(cls, collective_op: "inductor_ir.TensorBox"):
-    collective_op.decide_layout()
-    return inductor_ir.Wait(
-        layout=inductor_ir.AliasedLayout(collective_op),  # type: ignore[arg-type]
-        inputs=[collective_op],
-    )
-
-
-inductor_ir.Wait.create = wait_create  # type: ignore[assignment]
-
-inductor_ir.AllReduce.get_mutation_names = lambda self: [self.inputs[0].get_name()]  # type: ignore[attr-defined]
-
-
-@classmethod
-def all_reduce_create(
-    cls,
-    x: "inductor_ir.TensorBox",
-    reduce_op: str,
-    tag: str,
-    ranks: List[int],
-    group_size: int,
-):
-    inplace_inputs = cls.wrap_inputs_as_inplace([x])
-    layout = inductor_ir.MutationLayout(inplace_inputs[0])
-
-    _ = inductor_ir.AllReduce(
-        layout=layout,
-        inputs=inplace_inputs,
-        constant_args=[tag, ranks, group_size],
-        reduce_op=reduce_op,
-    )
-    return inplace_inputs[0]
-
-
-inductor_ir.AllReduce.create = all_reduce_create  # type: ignore[assignment]
-
-
-def wcg_codegen_free(self, buffer):
-    name = buffer.get_name()
+def get_volatile_reads_fixed(self):
```

Comment:
this is new? is there some pytorch issue this fixes?

Reply:
yes, this fixes an issue with the new functional collectives they've added in PT 2.2 (and have been improving in the nightlies). I need to file it so they fix it in the main codebase, just been busy getting this to run first.

Reply:
ok, it's filed now https://github.com/pytorch/pytorch/issues/121311

Reply:
let's make sure there's a link here to the issue in the comments so it's easier to find when we need to remove this workaround